cmd/bldr/cmd/rehash.go

```diff
@@ -0,0 +1,87 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cmd
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(rehashCmd)
+}
+
+var rehashCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update pkgs",
+	Args:  cobra.NoArgs,
+	RunE: func(_ *cobra.Command, _ []string) error {
+		for _, d := range datas {
+			if err := doTemplate(
+				template.Must(template.New(d["url"]).Funcs(sprig.HermeticTxtFuncMap()).Parse(d["url"])),
+				d,
+			); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	},
+}
+
+// datas is a list of deps to update. Provided as an example.
+//
+//nolint:lll
+var datas = []map[string]string{
+	{"VERSION": "v1.30.3", "url": "https://github.com/kubernetes/cloud-provider-aws/archive/refs/tags/{{ .VERSION }}.tar.gz"},
```

Comment:
can we extract that automatically?

Reply:
let's move this to `bldr update` itself, i was thinking of doing something similar to renovate, have additional fields, something like:

```yaml
steps:
  - sources:
      - url: https://download.qemu.org/qemu-{{ .QEMU_VERSION }}.tar.xz
        destination: qemu.tar.xz
        sha256: a8c3f596aece96da3b00cafb74baafa0d14515eafb8ed1ee3f7f5c2d0ebf02b6
        sha512: 58ed84f6fe6263d279356bc9193f96edf62cf3663fb151daa3f047d52329fe49cb91c2d45e09697e0469f4f5409be96403aec9572d4871ffa40848a786c21599
        update_options:
            uri: github://qemu/qemu-guest-agent
            filter: <some filtering option>
      - url: https://download.nvidia.org/nvidia-{{ .NVIDIA_VERSION }}.tar.xz
        destination: nvidia.tar.xz
        sha256: a8c3f596aece96da3b00cafb74baafa0d14515eafb8ed1ee3f7f5c2d0ebf02b6
        sha512: 58ed84f6fe6263d279356bc9193f96edf62cf3663fb151daa3f047d52329fe49cb91c2d45e09697e0469f4f5409be96403aec9572d4871ffa40848a786c21599
        update_options:
            uri: url:download.nvidia.org/releases.json
            filter: <some filtering option> // maybe use cue to filter from json or use `.foo.bar` syntax
```

this was we can mostly get rid of renovate